src/differenceWith.js

```diff
@@ -11,7 +11,7 @@ var _curry3 = require('./internal/_curry3');
  * @memberOf R
  * @since v0.1.0
  * @category Relation
- * @sig (a -> a -> Boolean) -> [*] -> [*] -> [*]
+ * @sig ((a, a) -> Boolean) -> [a] -> [a] -> [a]
```

Comment:
The signature should actually be `((a, b) -> Boolean) -> [a] -> [b] -> [a]`, since the two lists needn't be of the same type. For example:

```javascript
> R.differenceWith((r, id) => r.id === id, [{id: 1}, {id: 2}, {id: 3}], [2, 4, 6])
[{id: 1}, {id: 3}]
```